crates/sovereign-sdk/module-system/sov-state/src/codec/borsh_codec.rs

```diff
@@ -41,43 +48,43 @@ impl StateValueCodec<AlloyU256> for BorshCodec {
     }
 }
 
-impl<T> StateKeyCodec<Vec<T>> for BorshCodec
-where
-    T: BorshSerialize,
-{
-    fn encode_key(&self, value: &Vec<T>) -> Vec<u8> {
-        borsh::to_vec(value).expect("Failed to serialize key")
+// This one is needed for PublicKey only.
+// FIXME: Remove before mainnet
+impl StateKeyCodec<Vec<u8>> for BorshCodec {
+    fn encode_key(&self, value: &Vec<u8>) -> Vec<u8> {
+        value.clone()
     }
 }
 
-impl<T> StateValueCodec<Vec<T>> for BorshCodec
-where
-    T: BorshSerialize + BorshDeserialize,
-{
+impl StateValueCodec<Vec<u8>> for BorshCodec {
     type Error = std::io::Error;
 
-    fn encode_value(&self, value: &Vec<T>) -> Vec<u8> {
-        borsh::to_vec(value).expect("Failed to serialize value")
+    fn encode_value(&self, value: &Vec<u8>) -> Vec<u8> {
+        value.clone()
     }
 
-    fn try_decode_value(&self, bytes: &[u8]) -> Result<Vec<T>, Self::Error> {
-        borsh::from_slice(bytes)
+    fn try_decode_value(&self, bytes: &[u8]) -> Result<Vec<u8>, Self::Error> {
+        Ok(bytes.to_vec())
     }
 }
 
 macro_rules! impl_borsh_codec {
     ($t:tt) => {
         impl StateKeyCodec<$t> for BorshCodec {
             fn encode_key(&self, value: &$t) -> Vec<u8> {
-                borsh::to_vec(value).expect("Failed to serialize key")
+                let mut buf = Vec::with_capacity(8);
+                BorshSerialize::serialize(value, &mut buf).unwrap();
+                buf
             }
         }
 
         impl StateValueCodec<$t> for BorshCodec {
             type Error = std::io::Error;
 
             fn encode_value(&self, value: &$t) -> Vec<u8> {
-                borsh::to_vec(value).expect("Failed to serialize value")
+                let mut buf = Vec::with_capacity(8);
+                BorshSerialize::serialize(value, &mut buf).unwrap();
+                buf
             }
```

Comment:
maybe 8 is too small? not sure where this is used exactly, but on the top off my head 64 or 128 seems a bit more appropriate

Reply:
you can take the capacity as 2nd arg into the macro for easily doing it for uint and int types. for string, you can manually implement it